### ISSUE TITLE
Duplicate channels are shown

### ISSUE THREAD

**Issue by parkingzone** [babgvant/Emby.MythTv#6]
Duplicate channels are shown in Guide view (http://192.168.1.10:8096/web/livetvguide.html) and Channels view (http://192.168.1.10:8096/web/livetvchannels.html).

I have two separate video sources defined in MythTV. I believe this is causing channels that appear in both video sources to be duplicated. Can these duplicates be omitted.

As you know, the MythTV backend determines which video source should be used for live tv or recordings. Duplicate channels can cause problems if/when the MythTV API gets the ability to provide live tv streaming in the future.

FYI: This issue existed in earlier versions of the Kodi MythTV plugin. The new version eliminates the duplicates and lets the backend select the available video source. This code is on github (https://github.com/janbar/pvr.mythtv).

By the way, I am not a developer - just an enthusiast. I very well may be misunderstanding this. :-)

**Comment by babgvant** [comment]
Please test latest release (same URL). I can't test this so LMK the result.

**Comment by parkingzone** [comment]
It almost worked!

I tested refreshing the guide data from within Emby server twice with a server restart between attempts.

Only one channel was duplicated. It was the lowest channel number value (504). If channels are processed in numerical order then it would have been the first duplicated channel.

Otherwise, it worked just as expected. Let me know if you need more information.

MythTV Plugin Version: 1.0.5722.38984

**Comment by babgvant** [comment]
I'm filtering the channels by name. Does 504a have a difference name from 504b?

**Comment by parkingzone** [comment]
Yes, I found a name difference. I use custom names for some of my channels due to very long "official" names that originate from SchedulesDirect.

SourceID1: Fox HD
SourceID2: FOX HD

I think another variable would be better for comparison since channel names can be customized and different across sources.

How about using CallSign for comparison? This value will be the same for all sources. It would be safer to convert to lower or upper case before the comparison.

**Comment by babgvant** [comment]
Done.

**Comment by parkingzone** [comment]
Sorry for the late response. I didn't realize that you had updated the binary after this change.

This is working perfectly. The correct number of channels is returned without any duplicates.
